### ISSUE TITLE
DynamoDbEnhancedClient uses expression with reserved words

### ISSUE THREAD

**Issue by brackxm** [aws/aws-sdk-java-v2#2507]
The DynamoDbEnhancedClient uses the attributes names in expressions even when they are reserved words.

## Describe the bug
putItem fails if a property annotated with @DynamoDbVersionAttribute has a name that is a reserved word

## Expected Behavior
putItem succeeds

## Current Behavior
software.amazon.awssdk.services.dynamodb.model.DynamoDbException: Invalid ConditionExpression: Attribute name is a reserved keyword; reserved keyword: counter

## Steps to Reproduce

```
@DynamoDbBean
public class Customer {
    private String id;
    private Integer counter;

    @DynamoDbPartitionKey
    public String getId() {
        return id;
    }

    public void setId(final String id) {
        this.id = id;
    }

    @DynamoDbVersionAttribute
    public Integer getCounter() {
        return counter;
    }

    public void setCounter(final Integer counter) {
        this.counter = counter;
    }
}
```
```
        Customer customer = new Customer();
        customer.setId("customer1");
        table.putItem(customer);
```

## Possible Solution
use expression attribute names

## Context
migration from v1 DynamoDBMapper

## Your Environment
* AWS Java SDK version used: 2.16.72
* JDK version used: 11.0.10

**Comment by debora-ito** [comment]
Hi @brackxm thank you for reaching out.

I see how we can improve DynamoDB Enhanced client's condition expression by using expression attribute names. I'm curious about your use case, are you migrating from Java SDK v1 DynamoDB Mapper? Does your DynamoDB table currently have a field named `counter`? I'm wondering if you'd encounter more problems for using  a reserved word, and if you can change it to another name to avoid such problems.

**Comment by brackxm** [comment]
Yes this is for a migration from the v1 sdk.
Yes the table has an attribute counter.
This cannot easily be changed, and i would prefer to upgrade to the v2 sdk without data refactoring.
A "workaround" is to use the normal non-enhanced client for this.

**Comment by debora-ito** [comment]
Understood. We'll work to add expression attribute names to avoid issues with reserved words.

**Comment by debora-ito** [comment]
@brackxm a fix was released in Java SDK version `2.17.233`. 

Please let us know if you have any follow-up questions.

**Comment by jshankarc** [comment]
I could see the error for `key` - "Invalid ConditionExpression: Attribute name is a reserved keyword; reserved keyword: key" and I am using `2.21.5` version
